### PR TITLE
Fix temp file issue and filename

### PR DESCRIPTION
**Pull request by nitinprabhu**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RPE-228

### Change description ###

- Filename passed to temp file is the input zip file and not the output pgp file.

- `File.createTempFile` appends random long number to file which we don't want.Common's IO lib to the rescue.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/encryption/PgpEncryptionUtil.java [hunk 1]
```diff
@@ -35,8 +36,7 @@ private PgpEncryptionUtil() {
 
     public static byte[] encryptFile(
         byte[] inputFile,
-        String fileNamePrefix,
-        String fileNameSuffix,
+        String fileName,
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/encryption/PgpEncryptionUtil.java]: is this the input file name or result file name?
Javadoc would be nice

File: src/test/java/uk/gov/hmcts/reform/sendletter/encryption/PgpEncryptionUtilTest.java [hunk 2]
```diff
@@ -21,14 +21,10 @@ public void should_encrypt_and_create_pgp_encrypted_zip_file_when_valid_public_k
...
 
-        String fileNamePrefix = "encrypted";
-        String fileNameSuffix = "pgp";
-
         //when
         byte[] pgpEncryptedZip = PgpEncryptionUtil.encryptFile(
             inputZipFile,
-            fileNamePrefix,
-            fileNameSuffix,
+            "encrypted.zip",
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/sendletter/encryption/PgpEncryptionUtilTest.java]: file name is different that the name of the file that we pass. is this how this class is going to be used in real life?

Comment (nitinprabhu) [reply src/test/java/uk/gov/hmcts/reform/sendletter/encryption/PgpEncryptionUtilTest.java]: you can use any file name in our real service we would be passing something similar to `CMC001_cmcclaimstore_22032018152746_d07c653b-9bcd-4c54-85ca-e22aa321d501.zip`

File: src/main/java/uk/gov/hmcts/reform/sendletter/encryption/PgpEncryptionUtil.java [hunk 3]
```diff
@@ -34,9 +34,24 @@
     private PgpEncryptionUtil() {
     }
 
+    /**
+     * Encrypts the given byte array using PGP encryption using Triple DES algorithm.
+     * This method assumes that the temporary volume is writable
+     *
+     * @param inputFile          input file byte array
+     * @param inputFileName      input file name to be encrypted
+     * @param pgpPublicKey       public key to encrypt
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/encryption/PgpEncryptionUtil.java]: encrypt with

File: src/main/java/uk/gov/hmcts/reform/sendletter/encryption/PgpEncryptionUtil.java [hunk 4]
```diff
@@ -33,10 +34,24 @@
     private PgpEncryptionUtil() {
     }
 
+    /**
+     * Encrypts the given byte array using PGP encryption using Triple DES algorithm.
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/encryption/PgpEncryptionUtil.java]: let's use AES if possible (separate PR obv)

**Pull request merged**